Octokit/Http/ReadOnlyPagedCollection.cs

```diff
@@ -28,7 +28,14 @@ public ReadOnlyPagedCollection(IApiResponse<List<T>> response, Func<Uri, Task<IA
             var nextPageUrl = _info.GetNextPageUrl();
             if (nextPageUrl == null) return null;
 
-            var response = await _nextPageFunc(nextPageUrl).ConfigureAwait(false);
+            var maybeTask = _nextPageFunc(nextPageUrl);
+
+            if (maybeTask == null)
```

Comment:
I feel so awful about this escape hatch. But making `_nextPageFunc()` currently returns a task to indicate it has results. There's no real equivalent here to indicate we should bail out, so I'm going down this path.

Thankfully tests caught some stupid things I'd assumed, like `null.ConfigureAwait(false)` is a dumb idea.